crates/duckdb/src/raw_statement.rs

```diff
@@ -110,6 +112,39 @@ impl RawStatement {
         }
     }
 
+    #[inline]
+    pub fn streaming_step(&self, schema: SchemaRef) -> Option<StructArray> {
+        if let Some(result) = self.duckdb_result {
+            unsafe {
+                let mut out = ffi::duckdb_stream_fetch_chunk(result);
+
+                if out.is_null() {
```

Comment:
Does `out` need to be deallocated with `ffi::duckdb_destroy_data_chunk(&mut out)` even if its null?

Reply:
no need if following this pattern down below
```
        if !self.ptr.is_null() {
            unsafe { ffi::duckdb_destroy_prepare(&mut self.ptr) };
        }
```